libs/langchain/langchain/document_loaders/parsers/pdf.py

```diff
@@ -263,24 +263,94 @@ def lazy_parse(self, blob: Blob) -> Iterator[Document]:
 
 class DocumentIntelligenceParser(BaseBlobParser):
     """Loads a PDF with Azure Document Intelligence
-    (formerly Forms Recognizer) and chunks at character level."""
+    (formerly Forms Recognizer). Returns Document with
+    pages or paragraphs, table headers, and rows."""
 
-    def __init__(self, client: Any, model: str):
+    def __init__(self, client: Any, model: str, split_mode: str):
         self.client = client
         self.model = model
+        self.split_mode = split_mode
 
     def _generate_docs(self, blob: Blob, result: Any) -> Iterator[Document]:
-        for p in result.pages:
```

Comment:
if split mode is page should we just keep existing logic? is there value in parsing by paragraph and re-assembling pages?

Reply:
@baskaryan the idea of providing paragraphs as an option is to do chunking (splitting) as supported by the azure AI cognitive layout capabilities rather than having to do chunking again using, let’s say a Text Splitter. This would be helpful for generating embeddings of chunks. We won’t reassemble the paragraphs back into pages if `paragraph` is used rather we will keep it the way Doc intel’s layout extracts it. If the user specifies `page` explicitly or just doesn’t pass the parameter at initialization then page will be defaulted and entire page text will be generated per page. Hope this makes sense.

Reply:
@baskaryan right, I am actually doing this here.

https://github.com/langchain-ai/langchain/pull/10431/files#diff-45e28d6333af778cc97e27ef8b9e54e812071de26d30b12bd09ed40114a53821R280

Reply:
Attaching a sample JSON output from a 2 page document extracted via `prebuilt-read` model.

[output.json.zip](https://github.com/langchain-ai/langchain/files/12639778/output.json.zip)